pkg/core/blockchain.go

```diff
@@ -2113,6 +2115,8 @@ func (bc *Blockchain) persist(isSync bool) (time.Duration, error) {
 		persisted int
 		err       error
 	)
+	bc.wgPersist.Add(1)
+	defer bc.wgPersist.Done()
```

Comment:
As we see from the panic message, it's `persist` who is responsible for logging after test shutdown:
```
go.uber.org/zap.(*Logger).Info(0x9e00dd4e257b9dc4?, {0x10ae318?, 0x1b31620?}, {0xc00741c000, 0x5, 0x5})
	/home/anna/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:220 +0x4e
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).persist(0xc0017f4580, 0x0)
	/home/anna/Documents/GitProjects/nspcc-dev/neo-go/pkg/core/blockchain.go:2119 +0x4be
```
Checking out to the appropriate commit allows us to detect the "faulty" lines:
```
		bc.log.Info("persisted to disk",
			zap.Uint32("blocks", diff),
			zap.Int("keys", persisted),
			zap.Uint32("headerHeight", storedHeaderHeight),
			zap.Uint32("blockHeight", bHeight),
			zap.Duration("took", duration))
```

Navigate to the persist caller, it's `(bc *Blockchain) Run()`. As we've discussed, the first thing we need to check is whether `(bc *Blockchain) Close()` waits for `Run()` to exit properly. Turns out that it waits with the help of `runToExitCh`:
```
		bc.isRunning.Store(false)
		close(bc.runToExitCh)
```
And thus, you don't need this syncing code.

After that, you have to check whether blockchain is properly closed in `TestFailedPreconditionShutdown`. Turns out that it's not:
https://github.com/nspcc-dev/neo-go/blob/8d67f17943553740fe4caa31c3e9efcf7921aba2/pkg/services/rpcsrv/server_test.go#L4138-L4141

Blockchain variable is not even named. So what you really need in this test is to properly close blockchain in the end of the test, that's all.